query/src/test/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchPropertyHelperTest.java

```diff
@@ -20,14 +21,14 @@
  */
 public class HibernateSearchPropertyHelperTest {
 
-   @Rule
-   public SearchFactoryHolder factoryHolder = new SearchFactoryHolder(TestEntity.class);
-
    private HibernateSearchPropertyHelper propertyHelper;
 
    @Before
    public void setup() {
-      propertyHelper = new HibernateSearchPropertyHelper(factoryHolder.getSearchFactory(), new ReflectionEntityNamesResolver(null));
+      SearchMappingHolder mappingHolder = mock(SearchMappingHolder.class);
+      when(mappingHolder.getSearchMapping()).thenReturn(
+            SearchMappingHelper.createSearchMappingForTests(TestEntity.class));
```

Comment:
@fax4ever Unless I'm mistaken, you're not closing the search mapping, which leads to the test failures in CI mentioning "leaked threads".

There are other test failures mentioning thread leaks, maybe you should have a look at them? The name of the test generally appears somewhere in the stack trace.

Reply:
Sure. They seems to be related to `LuceneTransformationTest` test instance.

Reply:
in this test we open and (close -- maybe not) a search mapping instance manually to check the Lucene queries derived from it

Reply:
> Sure. They seems to be related to LuceneTransformationTest test instance.

Actually they are not. If you look at the stack trace, the threads were created in other tests. `HibernateSearchPropertyHelperTest` in particular, but there may be others.